### PR TITLE
Workgroups2 layer

### PR DESCRIPTION
**Pull request by drewkett**
This adds basic support for a workgroups2 layer

### CODE REVIEW

**Comment by CestDiego** [comment]
There is support for perspectives which is kinda like this: https://github.com/syl20bnr/spacemacs/pull/2845

**Comment by robbyoconnor** [comment]
How is this different than eyebrowse?

File: layers/+window-management/workgroups2/packages.el [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+;;
+;;; License: GPLv3
+
+;; List of all packages to install and/or initialize. Built-in packages
+;; which require an initialization must be listed explicitly in the list.
+(setq workgroups2-packages
+    '(workgroups2))
+
+;; List of packages to exclude.
+(setq workgroups2-excluded-packages '())
```
Comment (robbyoconnor) [inline layers/+window-management/workgroups2/packages.el]: Kill this -- why is it even there..delete what's not needed.

**Comment by drewkett** [comment]
I like workgroups2 because it saves the configuration to a file that's reloaded after a restart. I couldn't tell if either perspectives or eyebrowse did that.

**Comment by CestDiego** [comment]
Perspectives does that....but it would be nice if you could integrate it
into eyebrowse..so that eyebrowse could be more like perspectives, except
for buffer isolation of course

**Comment by drewkett** [comment]
I have a couple questions then. 

Is the only difference between eyebrowse and perspectives that perspectives have buffers attached specifically to them? (besides the general interface differences)

For save/load functionality, how is that done? I can see that its possible to define a custom perspective in a file, but is there any way to just have it save the current set up and have it reloaded on start up?

**Comment by CestDiego** [comment]
Yes, perspectives let's you reload them at startup... for emacs < 24.3
perspectives even uses Workspaces2 to support this..but as emacs 24.4
it just uses builtin code. You could try and expand eyebrowse with
workgroups,  but that's gonna take a little longer....

**Comment by robbyoconnor** [comment]
I am in no way opposed to this...options are cool.

--Rob

**Comment by drewkett** [comment]
@CestDiego Where does it show how to make perspectives load at startup? I don't see any mention of it, even with a quick look at the code.

**Comment by CestDiego** [comment]
@drewkett it's currently in this PR, not yet merged https://github.com/syl20bnr/spacemacs/pull/2845

**Comment by drewkett** [comment]
@CestDiego Oh, I missed that. I'll try that out when I get the chance.

**Comment by robbyoconnor** [comment]
Please don't close this pr...I like it.

--Rob

**Comment by CestDiego** [comment]
If you like it put a ring on it. :)

**Comment by robbyoconnor** [comment]
I already did...in fact I already married it.

--Rob

**Comment by robbyoconnor** [comment]
Also one more thing: squash into one tidy commit ;)

**Comment by StreakyCobra** [comment]
@drewkett Can you:

1. Rebase it on `develop`
2. Update the `README` to follow the `core/templates/layer-README.template`
3. Use the `SPC l` prefix like other `windows-management` layers

This would help to discuss this one further :smile:

**Comment by drewkett** [comment]
I haven't been using this lately, since I was having some slowdowns with emacs and disabling this sped things up. I bet I just had too many buffers open or something, and this package was loading them all back up on restart. I can rebase if that's desired, but I haven't had the time to figure out the issue.

**Comment by syl20bnr** [comment]
Thank you for the PR!

Since we have an official solution for workgroups, a.k.a. Spacemacs layouts which are perps-mode and eyebrowse based, I close this issue.

**Pull request closed without merging**